internal.nix

```diff
@@ -274,6 +290,11 @@ rec {
             '';
           executable = true;
         };
+
+        sourceHashFunc = spec:
+          if spec.type == "github" then
```

Comment:
If the type is a mismatch we could log some internal error. Currently it always *must* be `github` so we could even use and assert here.